src/Core/Foundation/Version.php

```diff
@@ -242,12 +242,14 @@ private function versionCompare($version, $operator)
      */
     private function checkVersion($version)
     {
-        if (!is_string($version)) {
-            throw InvalidVersionException::mustBeAString();
-        }
-
-        if (!preg_match('/^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$/', $version)) {
-            throw InvalidVersionException::mustBeValidName();
+        if (!preg_match('~^\d+(\.\d+){0,}$~', $version)) {
```

Comment:
@PierreRambaud I think this regex breaks the compare functions if you add test cases with 1.2.3.4. I have a look in the evening, maybe i'm wrong

Reply:
The regex accepts:
```
11
1.2.3
1.2.4.4.5
12.13.15
0.21.3
1
```

but doesn't accepts:
```
.33
0.1.
0.e.2
e.666
1.2.3-rc1
1.2.3-beta
```

You can have a check on VersionTest.php if you need more information.

Reply:
@PierreRambaud As I wrote to you 2 weeks ago, this regex does not work because for version_compare() the version 1.0.0 is greater than the version 1.0. That was why I did not change the regex when you first mentioned it.

I added a test which fails but which should be successfull.

I wrote the class for version strings with four numbers (PS versions) and it won't work with 1.0 or 7.4.6, that is what it is meant for. I also wrote to you that I never found another _PS_VERSION string which has less than 4 numbers.

Reply:
I always forgot this language really really sucks sometimes ^^ I add a real check (found in many websites & php.net) and all your tests are now valid ;)